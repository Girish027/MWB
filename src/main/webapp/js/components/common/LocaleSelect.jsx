import React, { Component } from 'react';
import Request from 'react-http-request';
import _ from 'lodash';
import getUrl, { pathKey } from 'utils/apiUrls';

class LocaleSelect extends Component {
  constructor(props) {
    super(props);
    this.processResults = this.processResults.bind(this);
  }

  processResults(result) {
    const locales = !_.isNil(result) && !_.isNil(result.body) ? result.body.sort() : [];
    return (
      <select {...this.props.input}>
        <optgroup>
          {locales.map(locale => (
            <option
              value={locale}
              key={locale}
            >
              {locale}
            </option>
          ))}
        </optgroup>
      </select>
    );
  }

  render() {
    // TODO - Move fetch outside of the select control
    const fetchUrl = getUrl(pathKey.locales);
    return (
      <div>
        <label>{this.props.label}</label>
        <Request
          url={fetchUrl}
          method="get"
          accept="application/json"
          verbose
        >
          {
            ({/* error, */ result, loading }) => {
              if (loading) {
                return (<option value="">loading...</option>);
              }
              return this.processResults(result);
            }
          }
        </Request>
      </div>
    );
  }
}
export default LocaleSelect;
