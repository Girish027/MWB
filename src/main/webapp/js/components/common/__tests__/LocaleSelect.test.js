import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import toJSON from 'enzyme-to-json';
import ReactDOM from 'react-dom';
import LocaleSelect from 'components/common/LocaleSelect';


describe('<LocaleSelect />', () => {
  let wrapper;

  const getShallowWrapper = () => shallow(<LocaleSelect />);

  describe('Snapshots', () => {
    test('renders correctly', () => {
      wrapper = getShallowWrapper();
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('should call processResults', () => {
      wrapper = getShallowWrapper();
      wrapper.instance().processResults({ body: ['2332'] });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('should call processResults without body', () => {
      wrapper = getShallowWrapper();
      wrapper.instance().processResults({ bgcolor: ['232'] });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
