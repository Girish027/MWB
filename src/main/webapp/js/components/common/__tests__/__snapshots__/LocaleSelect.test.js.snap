// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocaleSelect /> Snapshots renders correctly 1`] = `
<div>
  <label />
  <Request
    accept="application/json"
    method="get"
    url="/nltools/private/v1/resources/locales"
    verbose={true}
  >
    <Component />
  </Request>
</div>
`;

exports[`<LocaleSelect /> Snapshots should call processResults 1`] = `
<div>
  <label />
  <Request
    accept="application/json"
    method="get"
    url="/nltools/private/v1/resources/locales"
    verbose={true}
  >
    <Component />
  </Request>
</div>
`;

exports[`<LocaleSelect /> Snapshots should call processResults without body 1`] = `
<div>
  <label />
  <Request
    accept="application/json"
    method="get"
    url="/nltools/private/v1/resources/locales"
    verbose={true}
  >
    <Component />
  </Request>
</div>
`;
